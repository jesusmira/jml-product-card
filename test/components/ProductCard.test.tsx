import React from 'react';
import renderer from 'react-test-renderer';
import { ProductCard } from '../../src/components';
import { product1 } from './data/products';

const { act } = renderer;

describe('Pruebas en <ProductCard/>', () => { 

    test('debe de mostrar el componente correctamente', () => { 
        
        const warpper = renderer.create(
            <ProductCard product={ product1}>
                {
                    () => (
                        <h1>Product Card</h1>
                    )
                }
            </ProductCard>
        )

        expect(warpper.toJSON()).toMatchSnapshot();


     });

    test('debe de incrementar el contador', () => { 

        const warpper = renderer.create(
            <ProductCard product={ product1}>
                {
                    ({count, increaseBy}) => (
                        <>
                            <h1>Product Card</h1>
                            <span>{count}</span>
                            <button onClick={()=> increaseBy(1)}></button>
                        </>
                    )
                }
            </ProductCard>
        );

        let tree = warpper.toJSON();
        
        act(()=> 
            (tree as any).children[2].props.onClick()
        )
        tree = warpper.toJSON();

        expect((tree as any).children[1].children[0]).toBe('1');        

     })

 })