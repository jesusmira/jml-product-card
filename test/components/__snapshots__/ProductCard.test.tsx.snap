// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <ProductCard/> debe de mostrar el componente correctamente 1`] = `
<div
  className="productCard undefined"
>
  <h1>
    Product Card
  </h1>
</div>
`;
